### PR TITLE
Update the template to be more educational &  to run out of the box

### PR DESCRIPTION
**Pull request by efvink**
*Adding trace information
*Adding default implementation that provides an example of a detector for catching alerts in log analytics workspaces

### CODE REVIEW

File: src/SDK/SmartDetectors.ProjectTemplate/MyAlert.cs [hunk 1]
```diff
@@ -13,8 +13,43 @@ public class MyAlert : Alert
...
+        [AlertPresentationProperty(AlertPresentationSection.Property, "Disk name", Order = 2, InfoBalloon = "The disk name")]
+        public string DiskName { get; set; }
+
+        /// <summary>
+        /// Gets the query that displays the percentage of free space over time
+        /// </summary>
+        [AlertPresentationProperty(AlertPresentationSection.Chart, "% of free space", InfoBalloon = "A chart presenting the percentage of free space over time")]
+        public string FreeSpaceQuery => $@"
+                Perf                
+                | where CounterName == '% Free Space' and Computer == '{this.ComputerName.Replace("'", "''")}' and InstanceName == '{this.DiskName.Replace("'", "''")}'
```
Comment (liadtal) [inline src/SDK/SmartDetectors.ProjectTemplate/MyAlert.cs]: >CounterName [](start = 24, length = 11)

ObjectName == 'LogicalDisk' and CounterName == ...

File: src/SDK/SmartDetectors.ProjectTemplate/MySmartDetector.cs [hunk 2]
```diff
@@ -1,34 +1,59 @@
...
+        public async Task<List<Alert>> AnalyzeResourcesAsync(AnalysisRequest analysisRequest, ITracer tracer, CancellationToken cancellationToken)
         {
+            tracer.TraceInformation("Analyzing the specified resources...");
+
+            // Get the Log Analytics client
+            ITelemetryDataClient logAnalyticsDataClient = await analysisRequest.AnalysisServicesFactory.CreateLogAnalyticsTelemetryDataClientAsync(new List<ResourceIdentifier>() { analysisRequest.TargetResources.First() }, cancellationToken);
+            
+            // Run the query 
+            string query = @"Perf
+                            | where CounterName == '% Free Space' and CounterValue < 90
```
Comment (liadtal) [inline src/SDK/SmartDetectors.ProjectTemplate/MySmartDetector.cs]: >CounterName [](start = 36, length = 11)

ObjectName == 'LogicalDisk' and CounterName == ...

File: src/SDK/SmartDetectors.ProjectTemplate/MySmartDetector.cs [hunk 3]
```diff
@@ -1,34 +1,59 @@
...
+                            | summarize avgValue = avg(CounterValue) by Computer, InstanceName, bin(TimeGenerated, 30m)
```
Comment (liadtal) [inline src/SDK/SmartDetectors.ProjectTemplate/MySmartDetector.cs]: >summarize [](start = 30, length = 9)

where Computer == '{analysisRequest.TargetResources.First().ResourceName}'

File: src/SDK/SmartDetectors.ProjectTemplate/MySmartDetector.cs [hunk 4]
```diff
@@ -1,34 +1,59 @@
...
+    using System.Data;
     using System.Linq;
     using System.Threading;
     using System.Threading.Tasks;
     using Microsoft.Azure.Monitoring.SmartDetectors;
 
     /// <summary>
     /// A specific implementation of a <see cref="ISmartDetector"/>.	
-    /// The default implementation always returns a single alert with default title.
+    /// The default implementation provides an example of a detector for catching alerts in log analytics workspaces
```
Comment (liadtal) [inline src/SDK/SmartDetectors.ProjectTemplate/MySmartDetector.cs]: >catching alerts in log analytics workspaces  [](start = 73, length = 44)

detecting alerts based on data in log analytics

File: src/SDK/SmartDetectors.ProjectTemplate/MySmartDetector.cs [hunk 5]
```diff
@@ -1,34 +1,59 @@
...
-    /// A tracer used for emitting telemetry from the Smart Detector's execution. This telemetry will be used for troubleshooting and
-    /// monitoring the Smart Detector's executions.
-    /// </param>
-    /// <param name="cancellationToken">A <see cref="CancellationToken"/> to observe while waiting for a task to complete.</param>
-    /// <returns>
-    /// A <see cref="Task"/> that represents the asynchronous operation, returning the Alerts detected for the target resources. 
-    /// </returns>
-    public Task<List<Alert>> AnalyzeResourcesAsync(AnalysisRequest analysisRequest, ITracer tracer, CancellationToken cancellationToken)
+        /// <summary>
+        /// Initiates an asynchronous operation for analyzing the Smart Detector on the specified resources.
```
Comment (liadtal) [inline src/SDK/SmartDetectors.ProjectTemplate/MySmartDetector.cs]: >analyzing the Smart Detector [](start = 52, length = 28)

We are not analyzing the detector...
How about: "running the Smart Detector analysis on the specified resources"?

File: src/SDK/SmartDetectors.ProjectTemplate/MyAlert.cs [hunk 6]
```diff
@@ -13,8 +13,43 @@ public class MyAlert : Alert
...
+        /// Gets or sets the name of the disk
+        /// </summary>
+        [AlertPredicateProperty]
+        [AlertPresentationProperty(AlertPresentationSection.Property, "Disk name", Order = 2, InfoBalloon = "The disk name")]
+        public string DiskName { get; set; }
+
+        /// <summary>
+        /// Gets the query that displays the percentage of free space over time
+        /// </summary>
+        [AlertPresentationProperty(AlertPresentationSection.Chart, "% of free space", InfoBalloon = "A chart presenting the percentage of free space over time")]
```
Comment (liadtal) [inline src/SDK/SmartDetectors.ProjectTemplate/MyAlert.cs]: >% of free space [](start = 68, length = 15)

Let's add an example how to include other properties in the text: "% of free space in disk {DiskName}"

File: src/SDK/SmartDetectors.ProjectTemplate/MySmartDetector.cs [hunk 7]
```diff
@@ -1,34 +1,59 @@
...
+        /// </returns>
+        public async Task<List<Alert>> AnalyzeResourcesAsync(AnalysisRequest analysisRequest, ITracer tracer, CancellationToken cancellationToken)
         {
+            tracer.TraceInformation("Analyzing the specified resources...");
+
+            // Get the Log Analytics client
+            ITelemetryDataClient logAnalyticsDataClient = await analysisRequest.AnalysisServicesFactory.CreateLogAnalyticsTelemetryDataClientAsync(new List<ResourceIdentifier>() { analysisRequest.TargetResources.First() }, cancellationToken);
+            
+            // Run the query 
+            string query = @"Perf
```
Comment (liadtal) [inline src/SDK/SmartDetectors.ProjectTemplate/MySmartDetector.cs]: >Perf [](start = 29, length = 4)

Add "where TimeGenerated > ago(3h)"

File: src/SDK/SmartDetectors.ProjectTemplate/MySmartDetector.cs (hunk 3 above)
Comment (liadtal) [inline src/SDK/SmartDetectors.ProjectTemplate/MySmartDetector.cs]: >bin(TimeGenerated, 30m) [](start = 96, length = 23)

Doesn't make sense to summarize by time. Makes more sense to summarize by Computer+InstanceName, and then add: "where avgValue > 90"

File: src/SDK/SmartDetectorsSDK/ISmartDetector.cs [hunk 8]
```diff
@@ -16,7 +16,7 @@ namespace Microsoft.Azure.Monitoring.SmartDetectors
     public interface ISmartDetector
     {
         /// <summary>
-        /// Initiates an asynchronous operation for analyzing the Smart Detector on the specified resources.
+        /// Initiates an asynchronous operation for running the Smart Detector analysis on the specified resources.
```
Comment (EranGon) [inline src/SDK/SmartDetectorsSDK/ISmartDetector.cs]: >running [](start = 52, length = 7)

why?

File: src/SDK/SmartDetectors.ProjectTemplate/MyAlert.cs [hunk 9]
```diff
@@ -13,8 +13,33 @@ public class MyAlert : Alert
         /// </summary>
         /// <param name="title">The Alert's title.</param>
         /// <param name="resourceIdentifier">The resource identifier that this Alert applies to.</param>
-        public MyAlert(string title, ResourceIdentifier resourceIdentifier) : base(title, resourceIdentifier)
+        /// <param name="count">The number of records</param>
```
Comment (EranGon) [inline src/SDK/SmartDetectors.ProjectTemplate/MyAlert.cs]: >The number of records [](start = 32, length = 21)

The number of records found in the alert query

File: src/SDK/SmartDetectors.ProjectTemplate/MyAlert.cs (hunk 6 above)
Comment (EranGon) [reply src/SDK/SmartDetectors.ProjectTemplate/MyAlert.cs]: how about:
Number of records over time for {ResourceName}

---
In reply to: [183226901](https://github.com/liadtal/Smart-Alerts/pull/132#discussion_r183226901) [](ancestors = 183226901)

File: src/SDK/SmartDetectorsSDK/ISmartDetector.cs (hunk 8 above)
Comment (efvink) [reply src/SDK/SmartDetectorsSDK/ISmartDetector.cs]: @liadtal suggestion

File: src/SDK/SmartDetectors.ProjectTemplate/MyAlert.cs [hunk 10]
```diff
@@ -13,8 +13,33 @@ public class MyAlert : Alert
...
+        /// Gets or sets the number of records
+        /// </summary>
+        [AlertPresentationProperty(AlertPresentationSection.Property, "Number of records", Order = 2, InfoBalloon = "The number of records")]
+        public int Count { get; set; }
+
+        /// <summary>
+        /// Gets the query that displays the number of records over time
+        /// </summary>
+        [AlertPresentationProperty(AlertPresentationSection.Chart, "Number of records over time", InfoBalloon = "The number of records over time")]
+        public string FreeSpaceQuery => $@"
```
Comment (EranGon) [inline src/SDK/SmartDetectors.ProjectTemplate/MyAlert.cs]: >FreeSpaceQuery [](start = 22, length = 14)

CountChart

File: src/SDK/SmartDetectors.ProjectTemplate/MySmartDetector.cs [hunk 11]
```diff
@@ -1,34 +1,50 @@
...
+            // Process the query results and create alerts
             List<Alert> alerts = new List<Alert>();
-            alerts.Add(new MyAlert("title", analysisRequest.TargetResources.First()));
-            return Task.FromResult(alerts);
+            if (dataTables[0].Rows.Count > 0)
+            {
+                alerts.Add(new MyAlert("Title", analysisRequest.TargetResources.First(), Convert.ToInt32(dataTables[0].Rows[0]["Count"])));
+            }
+
+        tracer.TraceInformation("Finished creating alerts");
```
Comment (liadtal) [inline src/SDK/SmartDetectors.ProjectTemplate/MySmartDetector.cs]: >         [](start = 0, length = 8)

Indentation

File: src/SDK/SmartDetectors.ProjectTemplate/MySmartDetector.cs [hunk 12]
```diff
@@ -1,34 +1,50 @@
...
-    /// A tracer used for emitting telemetry from the Smart Detector's execution. This telemetry will be used for troubleshooting and
-    /// monitoring the Smart Detector's executions.
-    /// </param>
-    /// <param name="cancellationToken">A <see cref="CancellationToken"/> to observe while waiting for a task to complete.</param>
-    /// <returns>
-    /// A <see cref="Task"/> that represents the asynchronous operation, returning the Alerts detected for the target resources. 
-    /// </returns>
-    public Task<List<Alert>> AnalyzeResourcesAsync(AnalysisRequest analysisRequest, ITracer tracer, CancellationToken cancellationToken)
+        /// <summary>
+        /// Initiates an asynchronous operation for running the Smart Detector analysis on  the specified resources.
```
Comment (liadtal) [inline src/SDK/SmartDetectors.ProjectTemplate/MySmartDetector.cs]: >   [](start = 90, length = 2)

double space

**Pull request merged**